sdk/ml/azure-ai-ml/azure/ai/ml/entities/_inputs_outputs/input.py

```diff
@@ -225,6 +241,12 @@ def __init__(
             self.path = str(path)
         else:
             self.path = path
+        if path_on_compute is not None and not isinstance(path_on_compute, str):
+            # this logic will make dsl data binding expression working in the same way as yaml
+            # it's written to handle InputOutputBase, but there will be loop import if we import InputOutputBase here
```

Comment:
If you need to import `InputOutputBase` it should be possible to import it in this method or even if branch, by the time this code executes the import this module and the module containing `InputOutputBase` should already be initialized and come strait from the import cache.

Reply:
Thank you very much for your suggestions and I have checked that there should be no need to handle this. So, I have deleted it.